cmd/commands/run.go

```diff
@@ -94,15 +143,67 @@ func buildImg(ctx context.Context, fn function.Function) error {
 }
 
 // runFunction builds the function's images and runs the function.
-func runFunction(ctx context.Context, fn function.Function, eventName string) error {
-	if runSeed <= 0 {
-		rand.Seed(time.Now().UnixNano())
-		runSeed = rand.Int63n(1_000_000)
-	}
+func runFunction(ctx context.Context, fn function.Function, opts runFunctionOpts) error {
+	var evts []event.Event
+	var err error
 
-	evt, err := generateEvent(ctx, fn, eventName)
-	if err != nil {
-		return err
+	// We need to understand later if we're using a generated event so we can
+	// appropriately set a random seed if none has been provided.
+	usingGeneratedEvent := false
+
+	if opts.fetchEventId != nil {
+		// We're replaying a specific event, so let's fetch it.
+		evt, err := fetchEvent(ctx, *opts.fetchEventId)
+		if err != nil {
+			return err
+		}
+
+		if evt == nil {
+			return fmt.Errorf("event not found: %s", *opts.fetchEventId)
+		}
+
+		evts = []event.Event{*evt}
+	} else if opts.fetchRecentEvents > 0 {
```

Comment:
I was thinking about this and there's something we could do to improve the several if / else blocks chained together.

If we changed `runFunctionOpts` to:

```
type runFunctionOpts struct {
  eventFunc func ([]event.Event, error)
  verbose bool
}
```

When we initialize the flags, we could pass in a function which loads the events.  This `runFunction` code becomes much simpler — it's one single case of calling the evenFunc to return >= 1 event and an error.  Then, each `if` block becomes its own function that we can test individually, if desired.

It should make this area generally cleaner and a little easier to modify in the future.  WDYT?

Reply:
Agreed - would be much nicer!

Reply:
Have changed this to use a loading function for each scenario now.